web_console/api/job.js

```diff
@@ -13,54 +21,200 @@ const namespace = config.NAMESPACE;
+    },
+  };
 });
 
-router.get('/api/v1/job/:name/pods', SessionMiddleware, async (ctx) => {
-  const { name } = ctx.params;
-  const { pods } = await k8s.getFLAppPods(namespace, name);
-  ctx.body = { data: pods };
+router.get('/api/v1/job/:k8s_name/pods', SessionMiddleware, async (ctx) => {
+  const { k8s_name } = ctx.params;
+  const { pods } = await k8s.getFLAppPods(namespace, k8s_name);
+  ctx.body = { data: pods.items };
 });
 
-router.get('/api/v1/job/pod/:name/:container', SessionMiddleware, async (ctx) => {
-  const { name, container } = ctx.params;
+router.get('/api/v1/job/:k8s_name/logs/:start_time', SessionMiddleware, async (ctx) => {
+  const { k8s_name, start_time } = ctx.params;
+  const logs = await es.queryLog('filebeat-*', k8s_name, 'fedlearner-operator-*', start_time, Date.now(), es_oparator_match_phrase);
+  ctx.body = { data: logs };
+});
+
+router.get('/api/v1/job/pod/:pod_name/:container', SessionMiddleware, async (ctx) => {
+  const { pod_name, container } = ctx.params;
   const base = k8s.getBaseUrl();
-  const { id } = await k8s.getWebshellSession(namespace, name, container);
+  const { id } = await k8s.getWebshellSession(namespace, pod_name, container);
   ctx.body = { data: { id, base } };
 });
 
-router.get('/api/v1/job/pod/:name/logs/:time', SessionMiddleware, async (ctx) => {
-  const { name, time } = ctx.params;
-  const logs = await es.queryLog('filebeat-*', '', name, time, Date.now());
+router.get('/api/v1/job/pod/:pod_name/logs/:start_time', SessionMiddleware, async (ctx) => {
+  const { pod_name, start_time } = ctx.params;
+  const logs = await es.queryLog('filebeat-*', '', pod_name, start_time, Date.now());
   ctx.body = { data: logs };
 });
 
-router.post('/api/v1/job/create', SessionMiddleware, async (ctx) => {
-  const { fl_app } = ctx.request.body;
-  if (!fl_app) {
+// TODO: gRPC
+router.post('/api/v1/job', SessionMiddleware, async (ctx) => {
+  const { name, job_type, client_ticket_name, server_ticket_name } = ctx.request.body;
+
+  const [client_params_pass, client_params] = checkParseJson(ctx.request.body.client_params);
+  if (!client_params_pass) {
+    ctx.status = 400;
+    ctx.body = {
+      error: 'client_params must be json',
+    };
+    return;
+  }
+  const [server_params_pass, server_params] = checkParseJson(ctx.request.body.server_params);
+  if (!server_params_pass) {
     ctx.status = 400;
     ctx.body = {
-      error: 'Missing field: fl_app',
+      error: 'server_params must be json',
+    };
+    return;
+  }
+
+  const job = {
+    name, job_type, client_ticket_name, server_ticket_name,
+    client_params, server_params,
+  };
+
+  const exists = await Job.findOne({
+    where: {
+      name: { [Op.eq]: name },
+    },
+  });
+  if (exists) {
+    ctx.status = 422;
+    ctx.body = {
+      error: 'Job already exists',
     };
     return;
   }
-  const res = await k8s.createFLApp(namespace, fl_app);
-  ctx.body = { data: res };
+
+  const clientTicket = await Ticket.findOne({
```

Comment:
`Ticket` takes priority over `Job`, return error if ticket not exits, no need for job validation